lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -194,16 +194,60 @@ func (d *zfs) getDatasets(dataset string, types string) ([]string, error) {
 	return children, nil
 }
 
-func (d *zfs) setDatasetProperties(dataset string, options ...string) error {
+func (d *zfs) setDatasetProperties(dataset string, verifyCurrent bool, options ...string) error {
 	args := []string{"set"}
-	args = append(args, options...)
-	args = append(args, dataset)
 
-	_, err := shared.RunCommand("zfs", args...)
+	// Skip current property values verification.
+	if !verifyCurrent {
```

Comment:
As `setDatasetProperties` is now checking if the property matches the expected value, do we even need `verifyCurrent`?

In case the property is not yet there, the check below `currentProperties[keys[propertyIndex]] != values[propertyIndex]` would return false so the property gets set.

Reply:
this flag is there because a lot of the times the extra complexity of calling `getDatasetProperties` and checking the current values is not necessary. So `validateCurrent` makes the function more flexible and avoids doing that extra work when it is not needed. Without it I got some `timout` errors in the tests. Another option would be doing that verification on `ensureInitialDatasets` instead and leave `setDatasetProperties` unchanged, but I think it looks more clean this way.